### PR TITLE
Feature/10 record deserialization

### PR DESCRIPTION
**Pull request by apetrovs**

### CODE REVIEW

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/cluster/deserialization/SchemaRegistryRecordDeserializer.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+	private KafkaAvroDeserializer avroDeserializer;
+	private ObjectMapper objectMapper;
+	private StringDeserializer stringDeserializer;
+
+	private final Map<String, MessageFormat> topicFormatMap = new ConcurrentHashMap<>();
+
+	public SchemaRegistryRecordDeserializer(String schemaRegistryUrl) {
+		List<String> endpoints = Collections.singletonList(schemaRegistryUrl);
+		List<SchemaProvider> providers = Collections.singletonList(new AvroSchemaProvider());
+		this. schemaRegistryClient = new CachedSchemaRegistryClient(endpoints, 100, providers, Collections.emptyMap());
```
Comment (germanosin) [inline kafka-ui-api/src/main/java/com/provectus/kafka/ui/cluster/deserialization/SchemaRegistryRecordDeserializer.java:45]: What is `100` ? Please move magic numbers to constants or configs

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/cluster/deserialization/SchemaRegistryRecordDeserializer.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+			throw new RuntimeException("Failed to parse record from topic " + record.topic(), e);
+		}
+	}
+
+	private MessageFormat getMessageFormat(ConsumerRecord<Bytes, Bytes> record) {
+		return topicFormatMap.computeIfAbsent(record.topic(), k -> detectFormat(record));
+	}
+
+	private MessageFormat detectFormat(ConsumerRecord<Bytes, Bytes> record) {
+		String avroSchema = String.format(AVRO_SCHEMA_TEMPLATE, record.topic());
```
Comment (germanosin) [inline kafka-ui-api/src/main/java/com/provectus/kafka/ui/cluster/deserialization/SchemaRegistryRecordDeserializer.java:81]: Please move the template to per cluster config with default value

Comment (germanosin) [reply kafka-ui-api/src/main/java/com/provectus/kafka/ui/cluster/deserialization/SchemaRegistryRecordDeserializer.java:81]: And you can cache this state somewhere for later messages

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/cluster/deserialization/SchemaRegistryRecordDeserializer.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+			return MessageFormat.AVRO;
+		} catch (RestClientException | IOException e) {
+			log.info("Failed to get Avro schema for topic {}", record.topic());
+		}
+
+		try {
+			parseJsonRecord(record);
+			return MessageFormat.JSON;
+		} catch (IOException e) {
+			log.info("Failed to parse json from topic {}", record.topic());
```
Comment (germanosin) [inline kafka-ui-api/src/main/java/com/provectus/kafka/ui/cluster/deserialization/SchemaRegistryRecordDeserializer.java:93]: You may remember that it failed/passed before

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/cluster/deserialization/SchemaRegistryRecordDeserializer.java (hunk 1 above)
Comment (apetrovs) [reply kafka-ui-api/src/main/java/com/provectus/kafka/ui/cluster/deserialization/SchemaRegistryRecordDeserializer.java]: Changed.

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/cluster/deserialization/SchemaRegistryRecordDeserializer.java (hunk 2 above)
Comment (germanosin) [reply kafka-ui-api/src/main/java/com/provectus/kafka/ui/cluster/deserialization/SchemaRegistryRecordDeserializer.java:81]: It's rare but the possible case, please move it to the config

Comment (apetrovs) [reply kafka-ui-api/src/main/java/com/provectus/kafka/ui/cluster/deserialization/SchemaRegistryRecordDeserializer.java]: Ok, done.

**Pull request merged**